articles/tutorial-qdk-grovers-search.md

```diff
@@ -11,49 +11,367 @@ title: Run Grover's search algorithm in Q# - Quantum Development Kit
+> In this tutorial we are going to factorize a number using Grover's search algorithm as a didatic example to show how to translate a simple mathematical problem into a Grover's task. However, **Grover's algorithm is NOT an efficient algorithm to solve the integer factorization problem**. To explore a quantum algorithm that does solve the integer factorization problem faster than any classical algorithm check the [**Shor's algorithm** sample](https://github.com/microsoft/Quantum/tree/main/samples/algorithms/integer-factorization).
+
+As an example, let's see how we would express the function $f_M(x)=1[r]$ of the factoring problem as quantum operation in Q#.
+
+Classically, we would compute the rest of the division $M/x$ and check if it's equal to zero. If it is, the program outputs `1` and if it's not, the program outputs `0`. We need to:
+
+- Compute the remainder of the division.
+- Apply a controlled operation over the output bit so that it's `1` if the remainder is `0`.
+
+So we need to calculate a division of two numbers with a quantum operation. Fortunately, you don't need to write the circuit implementing the division from scratch, you can use [`DivideI`](xref:microsoft.quantum.arithmetic.dividei) operation of the Numerics library instead.
+
+If we look into the description of `DivideI` we see that it needs three qubit registers, the $n$-bit dividend `xs`, the $n$-bit divisor `ys` and the
+$n$-bit `result` that must be initialized in the state `Zero`. The operation is `Adj + Ctl`, so we can conjugate it and use it in *within-apply* statements. Also, in the description it says that the dividend in the input register `xs` is replaced by the remainder. This is perfect since we are interested exclusively in the remainder, and not in the result of the operation.
+
+We can build then a quantum operation that does the following:
+
+1. Takes three inputs:
+   - The dividend `number`: `Int`, this is the $M$ of $f_M(x)$.
+   - A qubit array encoding the divisor `divisorRegister : Qubit[]`, this is, $x$, that might be in a superposition state.
+   - A target qubit `target : Qubit` that flips if the output of $f_M(x)$ is $1$.
+1. Calculates the division $M/x$ using only reversible quantum operations, and flips the state of `target` if and only if the remainder is zero.
+1. Reverts all the operations except the flipping of `target` to leave the used auxiliary qubits in the zero state without introducing any irreversible operation like measurement. This step is important to preserve entanglement and superposition during the process.
+
+The code to implement this quantum operation is:
+
+```qsharp
+operation markingDivisor (
+    dividend : Int,
+    divisorRegister : Qubit [],
+    target : Qubit
+) : Unit is Adj+Ctl {
+    // Calculate the bit-size of the dividend.
+    let size = BitSizeI(dividend);
+    // Allocate two new qubit registers for the dividend and the result.
+    using ( (dividendQubits, resultQubits) = (Qubit[size], Qubit[size]) ){
```

Comment:
This uses 0.14 syntax rather than the new 0.15 syntax:
```suggestion
    use dividendQubits = Qubit[size];
    use resultQubits = Qubit[size];
```